### PR TITLE
Mconcat/stableswap bench

### PR DESCRIPTION
**Pull request by mconcat**
Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: x/gamm/pool-models/stableswap/amm.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ func cfmmConstant(xReserve, yReserve sdk.Dec) sdk.Dec {
 	x2 := xReserve.Mul(xReserve)
 	y2 := yReserve.Mul(yReserve)
 	return xy.Mul(x2.Add(y2))
+
```
Comment (alexanderbez) [inline x/gamm/pool-models/stableswap/amm.go:14]:
```suggestion
```

File: x/gamm/pool-models/stableswap/amm.go [hunk 2]
```diff
@@ -148,6 +149,39 @@ func solveCfmm(xReserve, yReserve, yIn sdk.Dec) sdk.Dec {
 	return a
 }
 
+func approx_eq(a, b, tol sdk.Dec) bool {
```
Comment (alexanderbez) [inline x/gamm/pool-models/stableswap/amm.go:152]:
```suggestion
func approxDecEqual(a, b, tol sdk.Dec) bool {
```

File: x/gamm/pool-models/stableswap/amm.go [hunk 3]
```diff
@@ -148,6 +149,39 @@ func solveCfmm(xReserve, yReserve, yIn sdk.Dec) sdk.Dec {
...
+	diff := a.Sub(b).Abs()
+	return diff.Quo(a).LTE(tol) && diff.Quo(b).LTE(tol)
+}
+
+var (
+	twodec, _    = sdk.NewDecFromStr("2.0")
```
Comment (alexanderbez) [inline x/gamm/pool-models/stableswap/amm.go:158]: use `MustNewDecFromStr` to avoid ignoring the error.

**Review by ValarDragon — APPROVED** [review]
LGTM once bez' comments are resolved!

**Comment by ValarDragon** [comment]
(also have to fix lint)

**Pull request merged**